src/configs/FirebaseConfig.js

```diff
@@ -59,6 +60,23 @@ const FirebaseConfig = (props) => {
     }
   };
 
+  const getUserProfileV2 = async (url) => {
+    if (url && typeof url === 'string') {
+      const userId = await getUserId();
+      const parts = url.split('/profile/');
+      const urlSplit = parts?.[1];
+      const paramSplit = urlSplit?.split('?userId=');
+      const username = paramSplit?.[0];
+      const targetUserId = paramSplit?.[1];
+      const data = {
+        other_id: targetUserId,
+        username,
+        user_id: userId
+      };
+      handleMovePage(USER, data);
+    }
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `getUserProfileV2` function assumes that the URL will always be in the correct format and does not handle potential errors or inconsistencies in the URL structure. For example, if the URL does not contain '/profile/' or '?userId=', the function may behave unexpectedly. It would be better to add error handling to ensure the URL is in the expected format before processing it.
```diff
  const getUserProfileV2 = async (url) => {
    if (url && typeof url === 'string') {
      const userId = await getUserId();
      const parts = url.split('/profile/');
      const urlSplit = parts?.[1];
      if (!urlSplit) throw new Error("Invalid URL format. Expected '/profile/'");
      const paramSplit = urlSplit?.split('?userId=');
      if (!paramSplit || paramSplit.length < 2) throw new Error("Invalid URL format. Expected '?userId='");
      const username = paramSplit?.[0];
      const targetUserId = paramSplit?.[1];
      const data = {
        other_id: targetUserId,
        username,
        user_id: userId
      };
      handleMovePage(USER, data);
    }
  };
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->